[MB-14323, MB-14296, MB-14208] audit history table migrations

mmayo-truss opened the pull request:
## JIRA tickets for this change [MB-14323](https://dp3.atlassian.net/browse/MB-14323) [MB-14296](https://dp3.atlassian.net/browse/MB-14296) [MB-14208](https://dp3.atlassian.net/browse/MB-14208)
 
## Summary
This PR is grouping the migrations together for multiple issues. 

MB-14323: We are adding to the ignored_cols array for multiple to remove duplicate entries on the move history table when a new item is added. 
Note that this will remove these ids from being tracked in the `changed_values` column of the audit history table. There are some foreign keys that need to be tracked, mostly with duty locations, so I omitted those from the ignored columns and commented that in the migration file. 
Second Note: this migration file looks bigger than it is. I decided to change the formatting of how we typically use the `add_audit_history_table` function in order to make it easier to see all of the ignored columns and make appropriate comments.

MB-14208: We are adding a the `user_uploads` table to be tracked on the audit history table. 

MB-14296: We are adding the `backup_contacts` table to be tracked on the audit history table.



## Setup to Run Your Code

<details>
<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

<!-- Fill out the next section as you see fit, these are just suggestions. -->

1. Access the milmove app as a service member.
2. Go through the process of creating a move (Make sure you use the correct GBLOC, Fort McCoy is a duty station that will show up easily for our normal local dev patterns)
3. When you create the shipment do not add a secondary pickup or a secondary delivery address
4. 🛑 Don't submit the move yet
5. Go back and edit address to add a secondary pickup address
6. 🛑 Don't submit the move yet
7. Go back again to edit address to add a secondary delivery address
8. Sign and submit the move.
9. Log into the office app as a services counselor and view the move history for the move just created.
10. You should see the secondary addresses displayed as their own rows on the move history table, but not any accompanying empty rows for the id being attached to the shipment.
11. You will NOT see any entries currently for the `user_uploads` or the `backup_contacts`, since the query has not been modified. But if you are running this locally you should see entries on a db viewer for both the orders upload and the creation of the backup contacts when you went through the creation of the move.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Have the Jira acceptance criteria been met for this change?

## Screenshots
BEFORE:
![image](https://user-images.githubusercontent.com/110134470/197251396-ecfee8d2-8df5-4035-b40c-b3d8cd7f9001.png)

AFTER
![image](https://user-images.githubusercontent.com/110134470/197251266-82a73d4c-d58c-4c35-a174-121286785416.png)


DB view of audit history table featuring backup_contacts and user_upload entries:
![image](https://user-images.githubusercontent.com/110134470/197245045-c54c1d27-7cd9-4d06-b964-f14bc93a5aeb.png)

robot-mymove commented:
<!--
  0 failure: 
  1 warning:  This PR does not ...
  1 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to unit tests, even though it affects app code.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:link: <a href="https://dp3.atlassian.net/browse/MB-14323">MB-14323</a>, <a href="https://dp3.atlassian.net/browse/MB-14296">MB-14296</a>, <a href="https://dp3.atlassian.net/browse/MB-14208">MB-14208</a></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8304c88b795869a7d637765597de0d2c6a8b6e98
</p>

juliemoon reviewed (approved):
@mmayo-truss this looks great! Thanks for getting this done!
![Screen Shot 2022-10-21 at 2 54 54 PM](https://user-images.githubusercontent.com/16272040/197269231-06699d7b-36f6-4c17-8342-45bb75f26a86.png)

afoon reviewed (approved):
<img width="1385" alt="Screen Shot 2022-10-21 at 2 50 11 PM" src="https://user-images.githubusercontent.com/28371460/197268577-2f25a360-c957-413e-bac3-3f8d91813456.png">
LGTM! Look at how much space was saved with the filtering 🥹.

kctruss reviewed (approved):
Looks good! I tested the migration on EXP and everything worked beautifully. Well done!

afoon commented on migrations/app/schema/20221020202612_create-and-modify-move-history-triggers-for-multiple-tables.up.sql:
```diff
@@ -0,0 +1,163 @@
+		'uploaded_orders_id',
+		'entitlement_id',
+		'uploaded_amended_orders_id'
+	] -- origin_duty_location_id and new_duty_location_id are fks but are utilized to display supplemental information
```
Should `rank` be omitted here too? It's also part of the `service_member` table and results on duplicate entries if rank is updated.
<img width="1473" alt="Screen Shot 2022-10-21 at 6 09 55 PM" src="https://user-images.githubusercontent.com/28371460/197295912-a49af5fa-02c0-41c1-93c9-b11502c46ca4.png">

mmayo-truss replied:
@afoon we can totally omit rank if we do not want to show the duplicate entries here. I was just focusing on foreign key ids, but this makes sense.

mmayo-truss commented:
I added `grade` (which correlates to `rank`) to the ignored cols for the `orders` table to hide the duplicated rank rows as seen in this PR #9295. I also updated the event names to "Updated profile" as per @juliemoon's instructions. 
![image](https://user-images.githubusercontent.com/110134470/197570133-b8791a0d-a711-4d25-9f84-8d9d50f4ac73.png)

kctruss reviewed (approved):
Approved (again) after new changes pushed. Thanks for all of this @mmayo-truss !

juliemoon commented:
Thanks @mmayo-truss I see that the rank is not getting duplicated and that the event name when a customer updates their rank shows up as **Updated profile**.
<img width="1582" alt="Screen Shot 2022-10-24 at 12 59 40 PM" src="https://user-images.githubusercontent.com/16272040/197583166-0e6a194b-197a-4619-9213-7975908aec1d.png">
